src/main/java/io/reactivex/internal/operators/flowable/FlowableGroupBy.java

```diff
@@ -78,13 +95,21 @@ protected void subscribeActual(Subscriber<? super GroupedFlowable<K, V>> s) {
 
         boolean outputFused;
 
-        public GroupBySubscriber(Subscriber<? super GroupedFlowable<K, V>> actual, Function<? super T, ? extends K> keySelector, Function<? super T, ? extends V> valueSelector, int bufferSize, boolean delayError) {
+        public GroupBySubscriber(Subscriber<? super GroupedFlowable<K, V>> actual, Function<? super T, ? extends K> keySelector,
+                Function<? super T, ? extends V> valueSelector, int bufferSize, boolean delayError, 
+                Function<? super Consumer<Object>, ? extends Map<K, Object>> mapFactory) throws Exception {
             this.actual = actual;
             this.keySelector = keySelector;
             this.valueSelector = valueSelector;
             this.bufferSize = bufferSize;
             this.delayError = delayError;
-            this.groups = new ConcurrentHashMap<Object, GroupedUnicast<K, V>>();
+            if (mapFactory == null) {
+                this.groups = new ConcurrentHashMap<Object, GroupedUnicast<K, V>>();
```

Comment:
I'm not a fan of this structure. These could be constructor arguments prepared before the `GroupBySubscriber` gets initialized. So the try-catch will only cover the non-null case only and the use of `GroupBySubscriber` can look like before:

```java
source.subscribe(new GroupBySubscriber<T, K, V>(s, keySelector, valueSelector, bufferSize, delayError, groups, evictedGroups));
```